lib/bundler/process_lock.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+module Bundler
+  class ProcessLock
+    def self.lock(bundle_path = Bundler.bundle_path)
+      lock_file_path = File.join(bundle_path, "bundler.lock")
+      File.open(lock_file_path, "w") do |f|
```

Comment:
Ah I am wrong, there is still a file handle open so it won't actually be deleted.